development/tools/jobs/tester/componentsuite.go

```diff
@@ -0,0 +1,145 @@
+		assert.False(t, job.SkipReport)
+		assert.True(t, job.Decorate)
+		assert.Equal(t, 10, job.MaxConcurrency)
+		assert.Equal(t, s.Repository, job.PathAlias)
+		AssertThatExecGolangBuildpack(t, job.JobBase, s.Image, s.workingDirectory())
+		AssertThatSpecifiesResourceRequests(t, job.JobBase)
+		AssertThatHasExtraRefTestInfra(t, job.JobBase.UtilityConfig, "master")
+		AssertThatHasPresets(t, job.JobBase, PresetDindEnabled, PresetDockerPushRepo, PresetGcrPush, PresetBuildPr)
+		job.RunsAgainstChanges(s.FilesTriggeringJob)
+	}
+}
+
+func (s *ComponentSuite) postMasterTest(jobConfig config.JobConfig) func(t *testing.T) {
+	return func(t *testing.T) {
+		job := FindPostsubmitJobByNameAndBranch(
+			jobConfig.Postsubmits[s.repositorySectionKey()],
+			s.jobName("post-master"),
+			"master",
+		)
+		require.NotNil(t, job)
+
+		assert.Equal(t, []string{"^master$"}, job.Branches)
+		assert.Equal(t, 10, job.MaxConcurrency)
+		assert.True(t, job.Decorate)
+		assert.Equal(t, s.Repository, job.PathAlias)
+		AssertThatHasExtraRefTestInfra(t, job.JobBase.UtilityConfig, "master")
+		AssertThatHasPresets(t, job.JobBase, PresetDindEnabled, PresetDockerPushRepo, PresetGcrPush, PresetBuildMaster)
+		job.RunsAgainstChanges(s.FilesTriggeringJob)
+		AssertThatExecGolangBuildpack(t, job.JobBase, s.Image, s.workingDirectory())
+	}
+}
+
+func (s *ComponentSuite) preReleaseTest(jobConfig config.JobConfig) func(t *testing.T) {
+	return func(t *testing.T) {
+		for _, currentRelease := range s.Releases {
+			t.Run(currentRelease.String(), func(t *testing.T) {
+				job := FindPresubmitJobByNameAndBranch(
+					jobConfig.Presubmits[s.repositorySectionKey()],
+					GetReleaseJobName(s.moduleName(), currentRelease),
+					currentRelease.Branch(),
+				)
+				require.NotNil(t, job)
+
+				assert.Equal(t, []string{currentRelease.Branch()}, job.Branches)
+				assert.False(t, job.SkipReport)
+				assert.True(t, job.Decorate)
+				assert.Equal(t, 10, job.MaxConcurrency)
+				assert.Equal(t, s.Repository, job.PathAlias)
+				assert.True(t, job.AlwaysRun)
+				AssertThatExecGolangBuildpack(t, job.JobBase, s.Image, s.workingDirectory())
+				AssertThatSpecifiesResourceRequests(t, job.JobBase)
+				AssertThatHasExtraRefTestInfra(t, job.JobBase.UtilityConfig, currentRelease.Branch())
+				AssertThatHasPresets(t, job.JobBase, PresetDindEnabled, PresetDockerPushRepo, PresetGcrPush, PresetBuildRelease)
+				job.RunsAgainstChanges(s.FilesTriggeringJob)
+			})
+		}
+	}
+}
+
+func (s *ComponentSuite) componentName() string {
+	return path.Base(s.Path)
+}
+
+func (s *ComponentSuite) repositoryName() string {
+	return path.Base(s.Repository)
+}
+
+func (s *ComponentSuite) repositorySectionKey() string {
+	return strings.Replace(s.Repository, "github.com/", "", 1)
+}
+
+func (s *ComponentSuite) moduleName() string {
+	return fmt.Sprintf("%s-%s", s.repositoryName(), strings.Replace(s.Path, "/", "-", -1))
+}
+
+func (s *ComponentSuite) jobConfigPath() string {
```

Comment:
Why do we need the pointer on these functions, when we don't need it on the `GenericComponentSuite` ?

Reply:
We don't